code/game/objects/items/stacks/stack.dm

```diff
@@ -341,6 +344,28 @@
 	else
 		return ..()
 
+//Verb to split stacks
+/obj/item/stack/verb/split_verb()
+	set src in view()
+	set name = "Split"
+
+
+	if (!Adjacent(usr))
+		usr << SPAN_WARNING("You need to be in arm's reach for that!")
+		return
+
+	var/quantity = input(usr,
+	"This stack contains [amount]/[max_amount]. How many would you like to split off into a new stack?\n\
+	The new stack will be put into your hands if possible", "Split Stack", round(amount * 0.5)) as num
+
+	var/obj/item/stack/S = split(round(quantity, 1))
+	if (istype(S))
+		//Try to put the new stack into the user's hands
+		if (!(usr.put_in_hands(S)))
+			//If that fails, leave it beside the original stack
+			S.forceMove(loc)
```

Comment:
1) `put_in_hands()` already drop item on floor under mob if it fails to equip item in hands
2) If mob use that verb while holding stack in hands new stack can stuck in him.